hw/ip/flash_ctrl/doc/_index.md

```diff
@@ -65,40 +75,130 @@ The physical controller supports the following features
 *  Data and informational paritions within each bank of flash memory
 *  Arbitration between host requests and controller requests at the bank level
    *  Host requests are always favored, however the controller priority can escalate if it repeatedly loses arbitration
-   *  Since banks are arbitrated independently, where transactions may take different amounts of times to complete, the physical controller is also responsible for ensuring in-order response to both the controller and host.
+   *  Since banks are arbitrated independently and transactions may take different amounts of times to complete, the physical controller is also responsible for ensuring in-order response to both the controller and host.
 *  Flash read stage
-   *  Each bank maintains a parameterizable number of read buffers in front of the flash memory
+   *  Each bank maintains a parameterizable number of read buffers in front of the flash memory (default to 4)
    *  The read buffers behave as miniature read-only-caches to store flash data when flash words are greater than bus words.
+   *  When a program or erase collides with an entry stored in the read buffer, the buffer contents are invalidated.
 *  Flash program stage
-   *  Flash data work packing when flash word size is an integer multiple of bus word size.
+   *  Flash data word packing when flash word size is an integer multiple of bus word size.
+*  Flash scrambling
+   * Flash supports XEX scrambling using the prince cipher
 
 Features to be implemented
 
 *  Flash scrambling
+   * Scrambling is optional based on page boundaries and is configurable by software
 *  Flash ECC
+   * Flash supports SECDED on the flash word boundary, the ECC bits are stored in the metadata bits and are not normally visible to software.
+   * A feature is under consideration to expose the metadata bits to the flash protocol controller.
+   * ECC is optional based on page boudaries and is configurable by software
+
+### Flash Memory Overview
+
+Unlike sram, flash memory is not typically organized as a contiguous block of genertic storage.
+Instead it is organized into data partitions and information partitions.
+
+The data partition holds generic data like a generic memory would.
+The information partition typically holds metadata about the data partition.
+This includes but is not limited to information such as
```

Comment:
nit: perhaps change this to "This includes but is not limited to:" ('information' repeated here and bullets)

Reply:
fixed.